src/zoid/buttons/util.js

```diff
@@ -357,3 +360,60 @@ export function getButtonSize(
     }
   }
 }
+
+export const getModal: (
+  clientID: string,
+  merchantID: $ReadOnlyArray<string> | void
+) => Object = memoize(async (clientID, merchantID) => {
+  try {
+    const namespace = getNamespace();
+    if (!window[namespace].MessagesModal) {
+      const modalBundleUrl = () => {
+        let envPiece;
+        switch (getEnv()) {
+          case "test":
+            return "/base/test/integration/windows/button/modal.js";
+          case "local":
+          case "stage":
+            envPiece = "stage";
+            break;
+          case "sandbox":
+            envPiece = "sandbox";
+            break;
+          case "production":
+          default:
+            envPiece = "js";
+        }
+        return `https://www.paypalobjects.com/upstream/bizcomponents/${envPiece}/modal.js`;
+      };
+
+      // eslint-disable-next-line no-restricted-globals, promise/no-native
+      await new Promise((resolve, reject) => {
+        const script = document.createElement("script");
+        script.setAttribute("data-pp-namespace", namespace);
+        script.src = modalBundleUrl();
+        script.addEventListener("error", (err: Event) => {
+          reject(err);
+        });
+        script.addEventListener("load", () => {
+          document.body?.removeChild(script);
+          resolve();
+        });
+        document.body?.appendChild(script);
+      });
+    }
+
+    return window[namespace].MessagesModal({
+      account: `client-id:${clientID}`,
+      merchantId: merchantID?.join(",") || undefined,
+    });
+  } catch (err) {
```

Comment:
If there's an error scenario (e.g. the `modal.js` URL failed for some reason) we likely don't want that to be cached via the `memoize` wrapper. We'd like a retry to occur when the user clicks on the message again. It looks like the `memoize` util has a [reset capability](https://github.com/krakenjs/belter/blob/449034e63bca445708b62d9e08f7190a44ba4169/src/util.js#L295) we may be able to call to reset the memoized cache in the error scenario.

``` js
} catch (err) {
    getModal.reset();
    // ...
}
```

Reply:
VSCode forsees a problem with that code as identified by flow:
> Cannot call `getModal.reset` because property `reset` is missing in  statics of function type [1].Flow(prop-missing)

I've been searching in [PayPal's Github for examples of other implementations of that reset method](https://github.paypal.com/search?l=JavaScript&p=6&q=memoize+reset&type=Code), but haven't had any luck.  Closest I could find was [an implementation](https://github.paypal.com/Checkout/xo-buttonjs/blob/4495d4ae6d7aa28488379408f00d2dd5e54f357a/public/js/inlineGuest/creditCardForm.js) on a [handwritten memoize function](https://github.paypal.com/Checkout/xo-buttonjs/blob/4495d4ae6d7aa28488379408f00d2dd5e54f357a/public/js/button/util.js#L24).